### DISCUSSION TITLE
WSI resolution/magnification level

### DISCUSSION THREAD

**Discussion by vladpopovici** [Project-MONAI/MONAI#3995] [Q&A] [answered]
Hello,

Thanks for this great framework!
I would like to be able to read a specific level from a whole slide image, either by indicating the resolution (e.g. microns-per-pixel) or magnification/pyramid level. I looked at monai.apps.pathology.data dataset classes but couldn't find the appropriate parameters. Did I miss something?

Thanks a lot,
Vlad

**Comment by Nic-Ma** [comment]
Hi @drbeh ,

Could you please help share some comments here?

Thanks in advance.

**Comment by bhashemian** [comment, marked as answer]
Hi @vladpopovici, thanks for your feedback. 
Currently `WSIReader` (the main class to read images) and `LoadImage` transform both support reading the whole slide at a given level; however, `PatchWSIDataset` does not have this parameter and reads only at `level=0`. We will implement this feature to `PatchWSIDataset` as soon as possible. Thanks for brining this to our attention.

**Reply by bhashemian** [reply]
I have created issue #4034 to address this feature request.

**Comment by vladpopovici** [comment]
Hi both,

Thanks for looking into this matter.
Since you are at it, as a suggestion, maybe the specification of the level could also be allowed in terms of desired resolution (selecting the closest available in the pyramid). E.g. the user would like to access the level corresponding to 0.25 mpp (microns-per-pixel) from a pyramid with (level=0, 0.23 mpp) and (level=1, 0.46 mpp), etc. So PatchWSIDataset would iterate over level=0.

All the best,
Vlad

**Reply by bhashemian** [reply]
That's a great point @vladpopovici but that requires to use metadata of the whole slide image and currently we have avoided that in this implementation, so it can't be achieved in this quick fix. We can consider this in the future improvements.

**Reply by bhashemian** [reply]
The PR to add support for `level` is submitted and will be merged soon. Once it is merged, you can use it in your application and please let us know if there is any issue. https://github.com/Project-MONAI/MONAI/pull/4036
